### ISSUE TITLE
multiple-bound-locations reports on `?Sized` which cannot be in the `where` clause

### ISSUE THREAD

**Issue by barafael** [rust-lang/rust-clippy#12640]
### Summary

The new-ish multiple-bound-locations lint fires on unfixable code.

### Lint Name

multiple-bound-locations

### Reproducer

I tried this code:

```rust
    pub fn remove<Q: ?Sized>(&mut self, k: &Q) -> Option<V>
    where
        K: Borrow<Q>,
        Q: Hash + Eq,
    {
        for i in 0..self.entries.len() {
            if self.entries[i].0.borrow() == k {
                return Some(self.entries.swap_remove(i).1);
            }
        }

        None
    }
```

taken from: https://github.com/tokio-rs/tokio/blob/035a968bddcadcbfeff02efa7ccf1f5d9b69f149/tokio-stream/src/stream_map.rs#L470-L482

I saw this happen:

```
warning: bound is defined in more than one place
   --> tokio-stream/src/stream_map.rs:470:19
    |
470 |     pub fn remove<Q: ?Sized>(&mut self, k: &Q) -> Option<V>
    |                   ^
...
473 |         Q: Hash + Eq,
    |         ^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#multiple_bound_locations
    = note: `#[warn(clippy::multiple_bound_locations)]` on by default
```

I expected to see this happen: No lint fired here, as long as `?Sized` cannot live in a `where`-clause.

### Version

```text
rustc 1.79.0-nightly (4fd4797c2 2024-04-03)
binary: rustc
commit-hash: 4fd4797c2654977f545c9a91e2aa4e6cdbb38919
commit-date: 2024-04-03
host: x86_64-unknown-linux-gnu
release: 1.79.0-nightly
LLVM version: 18.1.2
```

**Comment by barafael** [comment]
related: https://github.com/rust-lang/rust/issues/20503

**Comment by barafael** [comment]
i was wrong, pls ignore

**Comment by y21** [comment]
This is true for old versions of Rust (pre 1.15), and clippy currently doesn't respect the `msrv` configuration for this lint, so there's indeed a bug here when the MSRV is set to lower than that.

However, this has already been reported in #12370, and will be fixed by #12388, so more of a duplicate in that regard :)
